app/controllers/api/v1/questions_controller.rb

```diff
@@ -6,9 +6,28 @@ class QuestionsController < ApiController
       before_action :authenticate!
 
       def index
-        # Only retrieve public questions
-        questions = Question.where(private: false)
-        render status: :ok, json: questions, each_serializer: QuestionSerializer
+        # Retrieve all public questions
+        public_questions = Question.where(private: false)
+
+        result_questions = if search_params[:title].present?
+                             # Search for public questions that contain the partial 'title' string
```

Comment:
`Search for public questions that contain the partial 'title' string` to `Search for public questions that contain the partial title param string`

Reply:
https://github.com/SpicyCookies/rails-interview-project/pull/9/commits/207d82b437d995e9832203d999fbe36c48333f13